crates/sui-types/src/messages.rs

```diff
@@ -1502,6 +1513,7 @@ pub enum EntryArgumentErrorKind {
     ObjectKindMismatch,
     UnsupportedPureArg,
     ArityMismatch,
+    ReferenceMutabilityMismatch,
```

Comment:
Nit, this should be maybe something like `SharedObjectMutabilityMismatch`? As the error doesn't exist for other object ownerships

Reply:
The reason I named it this way is because there is already talk in sui-research about immutable references to owned objects as well. So I wanted more generic name

Reply:
I see... Maybe something like "ObjectMutabilityMismatch" or something? There already is an error for that with Immutable objects so maybe we could just combine that into something more general. 

The term reference here is a bit confusing just in that we sort of auto-borrow objects into references, and this error isn't about that but rather about a restriction applied to input objects that are then being misused.

Reply:
Renamed to `ObjectMutabilityMismatch`